rust-bins/src/bin/keygen.rs

```diff
@@ -138,10 +203,74 @@ fn output_possibly_encrypted<X: SerdeSerialize>(
 }
 
 fn handle_generate_ar_keys(kgar: KeygenAr) -> Result<(), String> {
-    let bytes_from_file = succeed_or_die!(fs::read(kgar.rand_input), e => "Could not read random input from provided file because {}");
-    if bytes_from_file.len() < 64 {
-        return Err("Provided randomness should be of size at least 64 bytes".to_string());
+    // Read word list and make sure it contains 2048 words.
+    let bip39_vec: Vec<_> = include_str!("data/BIP39English.txt")
+        .split_whitespace()
+        .collect();
+    if bip39_vec.len() != 2048 {
+        return Err("The BIP39 word list must contain 2048 words.".to_string());
+    }
```

Comment:
For a future version:
- Allow the user to select the word list or specify their own.
- Add a hash check to ensure that the correct BIP39English.txt is loaded.

Reply:
If I understand correctly, the list is embedded into the binary. The hash would thus only prevent one from tampering with the source code.

Reply:
I don't think we want to over-engineer this. It is essentially a one-off tool.